webapp/openapi_parser.py

```diff
@@ -0,0 +1,29 @@
+from urllib import request
+from yaml import load
+
+from yaml import Loader
+
+
+def parse_openapi(url: str):
+    definition = request.urlopen(url)
+    loaded_definition = load(definition, Loader)
+    tagged_definition = {}
+
+    for endpoint in loaded_definition["paths"]:
+        for method in loaded_definition["paths"][endpoint]:
+            if method != "parameters":
+                tag = loaded_definition["paths"][endpoint][method]["tags"][0]
+                if tag in tagged_definition:
+                    if {
+                        endpoint: loaded_definition["paths"][endpoint]
+                    } not in tagged_definition[tag]:
+                        tagged_definition[tag] = [
+                            *tagged_definition[tag],
+                            {endpoint: loaded_definition["paths"][endpoint]},
+                        ]
+                else:
+                    tagged_definition[tag] = [
+                        {endpoint: loaded_definition["paths"][endpoint]}
+                    ]
```

Comment:
You'll need to `from collections import defaultdict` then you can simplify all of this considerably:

```suggestion
    tagged_definition = defaultdict(list)

    for endpoint, methods in loaded_definition["paths"].items():
        for method, definition in methods.items():
            if method != "parameters":
                tag = definition["tags"][0]
                tagged_definition[tag].append({endpoint: methods})

    return tagged_definition
```

Reply:
Will need to modify this slightly since as written, duplicates are returned:

![image](https://user-images.githubusercontent.com/35104482/206470887-cd3d9df1-fbfb-4ec8-8f96-ffdca3d278cb.png)